vapi/esx/settings/clusters/vms/transition.go

```diff
@@ -0,0 +1,226 @@
+// © Broadcom. All Rights Reserved.
+// The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package vms
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/vmware/govmomi/vapi/cis/tasks"
+	"github.com/vmware/govmomi/vapi/esx/settings/clusters"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type clusterTransitionPath types.ManagedObjectReference
+
+const (
+	TransitionPath = clusters.BasePath + "/%s/vms/transition"
+)
+
+func (c clusterTransitionPath) String() string {
+	cid := types.ManagedObjectReference(c).Value
+	return fmt.Sprintf(TransitionPath, cid)
+}
+
+// ValidationResult contains fields that describe a validation result.
+type ValidationResult struct {
+	// Notifications associated with the validation.
+	Notifications rest.Notifications `json:"notifications"`
+}
+
+// VmSelectionType defines the different selection types for VM selection.
+type VmSelectionType string
+
+const (
+	// VmSelectionTypeVmExtraConfig selects System VMs that have a specific property configured in the VM
+	// extra configuration. The property has a
+	// key='com.vmware.vim.eam.selection'. The value represents a unique
+	// identifier used for VM selection and is provided by the client.
+	VmSelectionTypeVmExtraConfig VmSelectionType = "VM_EXTRA_CONFIG"
+)
+
+// VmSelectionSpec structure contains fields
+// to describe the criteria used to select System VMs to which an
+// AlternativeVmSpec configuration is applied.
+type VmSelectionSpec struct {
+	// SelectionType is the type for this VmSelectionSpec.
+	SelectionType VmSelectionType `json:"selection_type"`
+
+	// ExtraConfigValue is the unique VM extra configuration property value. The recommended usage is
+	// with an UUID.
+	//
+	// See VmSelectionTypeVmExtraConfig for more details.
+	ExtraConfigValue string `json:"extra_config_value"`
+}
+
+// TransitionSpec contains fields that describe the specification for transitioning a System VM Solution.
+type TransitionSpec struct {
+	// SourceCluster is the cluster to transition from.
+	SourceCluster types.ManagedObjectReference `json:"source_cluster"`
+
+	// Solution is the target desired solution specification in vLCM.
+	Solution *SolutionSpec `json:"solution"`
+}
+
+// EnableSpec contains fields that describe specification for enablement of EAM managed solution in vLCM.
+type EnableSpec struct {
+	// EamAgencyID is the identifier of the solution in EAM (EAM agency).
+	EamAgencyID string `json:"eam_agency_id"`
+
+	// Solution is the target desired solution specification in vLCM.
+	Solution *SolutionSpec `json:"solution"`
```

Comment:
All nillable fields should have the JSON tag `omitempty`.

Reply:
this is not nillable